xls/ir/function.h

```diff
@@ -144,8 +150,15 @@ class Function {
   absl::Status Accept(DfsVisitor* visitor);
 
   // Creates a clone of the function with the new name 'new_name'. Function is
-  // owned by the same package.
-  absl::StatusOr<Function*> Clone(absl::string_view new_name) const;
+  // owned by targt_package.  call_remapping specifies any function
+  // substitutions to be used in the cloned function, e.g. If call_remapping
+  // holds {funcA, funcB}, any references to funcA in the function will be
+  // references to funcB in the cloned function.
+  absl::StatusOr<Function*> Clone(
+      absl::string_view new_name,
+      absl::optional<Package*> targt_package = std::nullopt,
```

Comment:
Make this a plain pointer with an optional value of nullptr. An optional pointer has too many states: nullopt, nullptr, real value.

Reply:
Done.